lib/slack_main.rb

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env ruby
+
+require "dotenv"
+require "pry"
+require "httparty"
+require "terminal-table"
+
+require_relative "recipient"
+require_relative "channel"
+require_relative "user"
+require_relative "workspace"
+
+Dotenv.load
+
+CHANNEL_URL = "https://slack.com/api/channels.list"
+USER_URL = "https://slack.com/api/users.list"
+
+def ask(prompt)
+  print "#{prompt} > "
+  input = gets.chomp
+  return input
+end
+
+def menu
+  puts "\nMENU"
+  puts "1. List users"
+  puts "2. List channels"
+  puts "3. Select user"
+  puts "4. Select channel"
+  puts "5. Details (on selected user/channel)"
+  puts "6. Send message"
+  puts "7. Quit"
+
+  input = ask("What do you want to do?")
+  return input
+end
+
+def clear_selected(workspace)
+  workspace.selected = nil
+end
```

Comment:
I would say this is a demeter violation. Instead of saying to the `Workspace` "your `selected` is now `nil`", it would be cleaner to call a method to ask it to clear itself.

It might be that you could move this method completely into `Workspace`.